### PR TITLE
move all computeAt root domain map tests to a single test file

### PR DESCRIPTION
**Pull request by liqiangxl**
simple clean.

### CODE REVIEW

File: test/test_ca_root_domain_map.cpp
Comment (liqiangxl) [inline test/test_ca_root_domain_map.cpp:1]: No code change, just moved from `test_gpu1/3.cpp`

Comment (liqiangxl) [reply test/test_ca_root_domain_map.cpp:1]: Strictly speaking, the validation function has been updated to utilize an existing one.

**Review by jjsjann123 — APPROVED** [review]
Thanks a ton for the clean up.

**Comment by jjsjann123** [comment]
!build

**Comment by liqiangxl** [comment]
> Thanks a ton for the clean up.

Thanks for the suggestions. Following this clean up, I'll do a similar one for tests about persistent buffers.

**Comment by liqiangxl** [comment]
Hi @xwang233, do you know why `nvfuser-ci/build_wheel_stable_17 — Failing (unknown) after 2 minutes https://nv/e2E/83450796`? This PR is just a simple clean up.

**Comment by liqiangxl** [comment]
!build

**Comment by xwang233** [comment]
> Hi @xwang233, do you know why `nvfuser-ci/build_wheel_stable_17 — Failing (unknown) after 2 minutes https://nv/e2E/83450796`? This PR is just a simple clean up.

cc @jjsjann123 

```
#20 76.62 [118/291] Building CXX object CMakeFiles/codegen_internal.dir/csrc/multidevice/communication.cpp.o
#20 76.62 FAILED: CMakeFiles/codegen_internal.dir/csrc/multidevice/communication.cpp.o 
#20 76.62 /opt/rh/gcc-toolset-12/root/usr/bin/c++ -DNVFUSER_DISTRIBUTED -DTORCH_CUDA_BUILD_MAIN_LIB -DUSE_C10D_GLOO -DUSE_C10D_NCCL -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -I/workspace/Fuser/csrc -I/workspace/Fuser/build/include -I/workspace/Fuser/lib/dynamic_type/src -isystem /workspace/Fuser/third_party/gloo -isystem /workspace/Fuser/third_party/flatbuffers/include -isystem /usr/local/cuda/extras/CUPTI/include -isystem /usr/local/cuda/include -isystem /opt/python/cp310-cp310/lib/python3.10/site-packages/torch/include -isystem /opt/python/cp310-cp310/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -Wno-psabi -D_GLIBCXX_USE_CXX11_ABI=0 -O3 -DNDEBUG -std=gnu++17 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wno-unused-function -Werror -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT CMakeFiles/codegen_internal.dir/csrc/multidevice/communication.cpp.o -MF CMakeFiles/codegen_internal.dir/csrc/multidevice/communication.cpp.o.d -o CMakeFiles/codegen_internal.dir/csrc/multidevice/communication.cpp.o -c /workspace/Fuser/csrc/multidevice/communication.cpp
#20 76.62 /workspace/Fuser/csrc/multidevice/communication.cpp: In member function ‘virtual c10::intrusive_ptr<c10d::Work> nvfuser::Reduce::post(nvfuser::Communicator&, std::optional<nvfuser::CommunicatorBackend>)’:
#20 76.62 /workspace/Fuser/csrc/multidevice/communication.cpp:243:15: error: cannot convert ‘__gnu_cxx::__alloc_traits<std::allocator<at::Tensor>, at::Tensor>::value_type’ {aka ‘at::Tensor’} to ‘std::vector<at::Tensor>&’
#20 76.62   243 |         buf.at(0), params_.src_bufs.at(0), options);
#20 76.62       |         ~~~~~~^~~
#20 76.62       |               |
#20 76.62       |               __gnu_cxx::__alloc_traits<std::allocator<at::Tensor>, at::Tensor>::value_type {aka at::Tensor}
#20 76.62 In file included from /workspace/Fuser/csrc/multidevice/communication.cpp:10:
#20 76.62 /opt/python/cp310-cp310/lib/python3.10/site-packages/torch/include/torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:432:32: note:   initializing argument 1 of ‘c10::intrusive_ptr<c10d::Work> c10d::ProcessGroupNCCL::_reduce_oop(std::vector<at::Tensor>&, std::vector<at::Tensor>&, const c10d::ReduceOptions&)’
#20 76.62   432 |       std::vector<at::Tensor>& outputTensors,
#20 76.62       |       ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
```

**Comment by jjsjann123** [comment]
That build error looks like something I recently fixed...

Was there another revert on the torch API?!?!

**Comment by jjsjann123** [comment]
```
#20 76.62 /opt/python/cp310-cp310/lib/python3.10/site-packages/torch/include/torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:432:32: note:   initializing argument 1 of ‘c10::intrusive_ptr<c10d::Work> c10d::ProcessGroupNCCL::_reduce_oop(std::vector<at::Tensor>&, std::vector<at::Tensor>&, const c10d::ReduceOptions&)’
#20 76.62   432 |       std::vector<at::Tensor>& outputTensors,
#20 76.62       |       ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
```

For some strange reason, the macro guard is picking up the wrong code...

**Comment by jjsjann123** [comment]
```["torch@https://download.pytorch.org/whl/cu121/torch-2.2.1%```

so we just had a patch release, that's probably why.

**Comment by xwang233** [comment]
> `["torch@https://download.pytorch.org/whl/cu121/torch-2.2.1%`
> 
> so we just had a patch release, that's probably why.

Patch release shouldn't change API, right?

**Comment by jjsjann123** [comment]
oh they didn't change anything. our macro guard is broken with that though. https://github.com/NVIDIA/Fuser/blob/6717e9660c487daa05b350e0026bd3ea2c1a3acf/csrc/multidevice/communication.cpp#L240

**Comment by jjsjann123** [comment]
linking #1809

**Comment by jjsjann123** [comment]
seems to be failing tests as well: https://gitlab-master.nvidia.com/dl/pytorch/fuser-gh-mirror/-/jobs/83467685/raw

I wonder how this could be affected by your PR?!?! likely a broken main?

**Comment by jjsjann123** [comment]
looks like head is broken :fearful: 
I think this could be my doing with the propagation stuff..... I might have accidentally merged two PRs in parallel without ever having those going through CI together. :face_exhaling:

**Comment by jjsjann123** [comment]
I don't think there's anything risky with this PR. I'm merging it.

failing CIs are patched in other PRs #1811 #1809

**Pull request merged**